lib/Dialect/Torch/Transforms/DecomposeComplexOps.cpp

```diff
@@ -1390,7 +1397,10 @@ class DecomposeAtenMeanDimOp : public OpRewritePattern<AtenMeanDimOp> {
     Value dimList = op.dim();
     Value keepDim = op.keepdim();
     Value dtype = op.dtype();
-    Type outputType = op.getType();
+    BaseTensorType outputTensorType =
+        op.result().getType().cast<BaseTensorType>();
+    Type newOutputType = outputTensorType.getWithSizesAndDtype(
```

Comment:
This will upcast int32 or int64 type to f64 type, which is not the main motivation regarding this PR. We should cast when there is a need, i.e, check dtype and then upcast. We should rather create a static function upcast that checks the dtype and casts accordingly.

Reply:
I see for `aten.mean` the output dtype is limited to float and complex.